jax/_src/scipy/stats/_core.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2022 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+from functools import partial
+from typing import NamedTuple, Optional, Tuple
+
+import jax.numpy as jnp
+import scipy
+from jax import jit
+from jax._src.api import vmap
+from jax._src.numpy.lax_numpy import _check_arraylike
+from jax._src.numpy.util import _wraps
+from jax._src.typing import Array, ArrayLike
+from jax._src.util import canonicalize_axis
+
+ModeResult = namedtuple('ModeResult', ('mode', 'count'))
+
+@_wraps(scipy.stats.mode, lax_description="""\
+Currently the only supported nan_policy is 'propagate'
+""")
+@partial(jit, static_argnames=['axis', 'nan_policy'])
+def mode(a: ArrayLike, axis: Optional[int] = 0, nan_policy: str = "propagate") -> ModeResult:
+  _check_arraylike("mode", a)
+  x = jnp.asarray(a)
+  x = jnp.atleast_1d(x)
+
+  if nan_policy not in {"propagate", "omit", "raise"}:
```

Comment:
Use a list rather than a set here, because if a user passes a non-hashable value for `nan_policy` this membership check will raise an error.